akka-projection-grpc/src/main/scala/akka/projection/grpc/consumer/GrpcQuerySettings.scala

```diff
@@ -4,25 +4,90 @@
 
 package akka.projection.grpc.consumer
 
+import akka.annotation.ApiMayChange
+import akka.grpc.scaladsl.Metadata
+import akka.grpc.scaladsl.MetadataBuilder
 import com.typesafe.config.Config
+import akka.util.ccompat.JavaConverters._
 
+import java.util.Optional
+import scala.compat.java8.OptionConverters._
+
+@ApiMayChange
 object GrpcQuerySettings {
-  def apply(config: Config): GrpcQuerySettings =
-    new GrpcQuerySettings(config)
+  def apply(config: Config): GrpcQuerySettings = {
+    val streamId = config.getString("stream-id")
+    require(
+      streamId != "",
+      "Configuration property [stream-id] must be an id exposed by the streaming side (but was empty).")
```

Comment:
"streaming side" might be difficult to understand. "exposed by producer side, but was undefined on consumer side"?